atd-vzd/migrations/migration_noncr3_location_2023_08_02--1648.sql

```diff
@@ -0,0 +1,495 @@
+-- 1. Add generated column for non-CR3 crash locations
+ALTER TABLE atd_apd_blueform
+ADD COLUMN generated_location_id varchar 
+GENERATED ALWAYS AS (update_noncr3_location(case_id)) STORED;
+
+-- 2. Drop views that use location_id
+DROP VIEW IF EXISTS locations_with_crash_injury_counts;
+DROP VIEW IF EXISTS view_location_crashes_global;
+DROP VIEW IF EXISTS view_location_injry_count_cost_summary;
+
+-- 3. Drop materialized views that use location_id and branch from all_atd_apd_blueform
+DROP MATERIALIZED VIEW IF EXISTS all_crashes_off_mainlane;
+DROP MATERIALIZED VIEW IF EXISTS all_non_cr3_crashes_off_mainlane;
+DROP MATERIALIZED VIEW IF EXISTS all_atd_apd_blueform;
+
+-- 4. Drop materialized views that use location_id and branch from five_year_atd_apd_blueform
+DROP MATERIALIZED VIEW IF EXISTS five_year_highway_polygons_with_crash_data;
+DROP MATERIALIZED VIEW IF EXISTS five_year_all_crashes_outside_any_polygons;
+
+DROP MATERIALIZED VIEW IF EXISTS five_year_all_crashes_outside_surface_polygons;
+DROP MATERIALIZED VIEW IF EXISTS five_year_non_cr3_crashes_outside_surface_polygons;
+
+DROP MATERIALIZED VIEW IF EXISTS five_year_surface_polygons_with_crash_data;
+DROP MATERIALIZED VIEW IF EXISTS five_year_all_crashes_off_mainlane_outside_surface_polygons;
+
+DROP MATERIALIZED VIEW IF EXISTS five_year_all_crashes_off_mainlane;
+DROP MATERIALIZED VIEW IF EXISTS five_year_non_cr3_crashes_off_mainlane;
+
+DROP MATERIALIZED VIEW IF EXISTS five_year_atd_apd_blueform;
+
+-- 5. Then, drop current atd_apd_blueform location_id column
+ALTER TABLE atd_apd_blueform DROP COLUMN location_id;
+
+-- 6. Rename generated_location_id to location_id to replace the previous column
+ALTER TABLE atd_apd_blueform 
+RENAME COLUMN generated_location_id TO location_id;
+
+-- 7. Add DB views that use location_id (need to be tracked in Hasura console)
+CREATE OR REPLACE VIEW locations_with_crash_injury_counts AS
```

Comment:
just checkin - did you copy the code from these views from the read replica or from the codebase? just thinking about how sometimes the gh version controlled db stuff isnt accurate anymore :face_with_spiral_eyes:

Reply:
great thinking - these did come from the production replica. just double-checked to be sure 🙏